### PR TITLE
Sweden - Node-based to cell area-based product

### PR DESCRIPTION
**Pull request by RohdeBSH**
To change the product specification from being Node- to Cell area-based is possible based on
functionality/metadata/attributes currently available in S-102 PS and S-100 ed. 5.0.0. Our suggestion (in dialog with
S-100) is changing the encoding in section 10.2.4. Root BathymetryCoverage more specifically the attributes of the
BathymetryCoverage feature container group.

### CODE REVIEW

**Comment by tfilppula** [comment]
Terminology needs further work and some clarifications are needed. Do not merge yet.

File: sources/3.0.0/sections/12-metadata.adoc [hunk 1]
```diff
@@ -401,7 +401,7 @@ _False_ indicates the dataset *is* intended to be used for navigation.
 
 |Attribute
 |boundingBox
-|The extent of the datast limits
+|The extent of the dataset limits defined as a rectangle coincident with the outermost cell boundaries of the dataset. NoData cells included.
```
Comment (tfilppula) [inline sources/3.0.0/sections/12-metadata.adoc:404]: I think there is no need to explicitly state that NoData/fill value cells must be included as, at least to me, the "outermost cell boundaries" implies this. I did however leave this in place in case there are differing opinions.

Comment (AnnaWall01) [reply sources/3.0.0/sections/12-metadata.adoc:404]: SMA agree. It does not have to be stated that NoData cells included. It is enough with "outermost cell boundaries".

File: sources/3.0.0/sections/10-data_product_format.adoc [hunk 2]
```diff
@@ -594,7 +594,7 @@ unsigned +
 | commonPointRule
 | 1
 | Enumeration
-| Value: 1 (average) or other values from <<iho-s100,part-10c,table=20>>.
+| Value: 1-4 <<iho-s100,part-10c,table=20>>.
```
Comment (tfilppula) [inline sources/3.0.0/sections/10-data_product_format.adoc:597]: A thought: should this be fixed to value 2 ("Use the least of the attribute values")?
I think this might be sensible, since we're talking about a navigation product. Value 2 would keep things on the safe side.
Maybe this is a question that should be posted as an issue for discussion?

Comment (AnnaWall01) [reply sources/3.0.0/sections/10-data_product_format.adoc:597]: SMA strongly support value 2. However, there might be others who prefer the other alternatives. As you say, a new issue might be suitable.

File: sources/3.0.0/sections/12-metadata.adoc (hunk 1 above)
Comment (tfilppula) [reply sources/3.0.0/sections/12-metadata.adoc:404]: Made a change accordingly.

File: sources/3.0.0/sections/04-data_content_struct.adoc [hunk 3]
```diff
@@ -229,7 +229,13 @@ In S-102 there are currently no complex attributes defined.
 === Dataset types
 
 ==== Introduction
-Bathymetric Surface datasets are represented as a discrete array of points contained in a regular grid. The general structure for a regular grid is defined in <<iho-s100,part=8>>.
+Bathymetric Surface datasets are represented as regular grids. The general structure for a regular grid is defined in <<iho-s100,part=8>>.
+
+The geo-referencing for an S-102 Bathymetric Surface product shall be node-based, referenced from the southwestern-most node in a grid. Each sample in a grid represents the value in the grid cell. The reference position included in the metadata shall be given in the coordinates used for the grid and shall contain sufficient digits of precision to locate the grid with accuracy no worse than a decimetre on the surface of the ellipsoid of rotation of the chosen horizontal datum.
+
+[[fig-s102-cell-area-and-Bbox]]
+.Cell Geometry
```
Comment (tfilppula) [inline sources/3.0.0/sections/04-data_content_struct.adoc:237]: This probably needs some more text.
Also this part of the document still has many occasions that use "grid node" or "data point" instead of "grid cell".

File: sources/3.0.0/sections/10-data_product_format.adoc (hunk 2 above)
Comment (tfilppula) [reply sources/3.0.0/sections/10-data_product_format.adoc:597]: Posted a new issue: https://github.com/iho-ohi/S-102-Product-Specification/issues/74

File: sources/3.0.0/sections/04-data_content_struct.adoc (hunk 3 above)
Comment (tfilppula) [reply sources/3.0.0/sections/04-data_content_struct.adoc:237]: I'll look into caption next week.

File: sources/3.0.0/sections/10-data_product_format.adoc [hunk 4]
```diff
@@ -639,10 +639,17 @@ For example, "latitude,longitude". Reverse scan direction along an axis is indic
...
 | Enumeration
-| Code value from <<iho-s100,part=10c,table=22>>
+| Value: 1. See code value from <<iho-s100,part=10c,table=22>>
+
+| 9
+| Offset of data point in cell
+| dataOffsetCode
+| 1
+| Enumeration
+| Value: 5 barycenter (centroid) of cell. See code value from <<iho-s100,part=10c,table=10>>
```
Comment (RohdeBSH) [inline sources/3.0.0/sections/10-data_product_format.adoc:652]: Why are you bringing this in?
It's an optional attribute that we didn't have before.
What bothers me is the value "Barycenter (centroid)".
From a geometric point of view, the centroid is not identical to the center point.

The center point and the centroid of a plane surface are two different concepts:

- Center point of a plane surface:
        The center point of a plane surface is the point that lies exactly in the middle of the surface. It is the geometric center that divides the surface into two symmetrical halves.

- Centroid of a plane surface:
        The centroid of a plane surface is the point where the entire mass or weight of the surface is concentrated. It is the point where the surface is in balance when it is balanced on a tip.


In our use case, where the cells are squares, the points are identical. 
The question is, what do you want? The center point or the centroid of the surface. If you choose the barycenter (centroid) here (I'm fine with that), it must also match the rest of the text.

In my opinion, the value 6 (center point) is missing in the enumeration.
That's why I said in Singapore: Leave it out, it's optional in the S-100 anyway.

File: sources/3.0.0/sections/12-metadata.adoc [hunk 5]
```diff
@@ -401,7 +401,7 @@ _False_ indicates the dataset *is* intended to be used for navigation.
 
 |Attribute
 |boundingBox
-|The extent of the datast limits
+|The extent of the dataset limits defined as a rectangle coincident with the outermost cell boundaries of the dataset.
```
Comment (RohdeBSH) [inline sources/3.0.0/sections/12-metadata.adoc:404]: How many data set limits are there?
One, right? 
Shouldn't it be "limit" instead of "limits"?

File: sources/3.0.0/sections/10-data_product_format.adoc (hunk 2 above)
Comment (tfilppula) [reply sources/3.0.0/sections/10-data_product_format.adoc:597]: I beg to differ.
You can query a grid at exactly at border of two (or three or four) grid cells containing data. If this happens, you have several nearest neighbours, so which one to report? I think the shoalest value makes most sense.

File: sources/3.0.0/sections/04-data_content_struct.adoc [hunk 6]
```diff
@@ -101,7 +101,7 @@ The attribute _axisNames_ has the value class _Sequence<CharacterString>_ that s
 [level=7]
 ===== extent
 
-The attribute _extent_ has the value class *CV_GridEnvelope* that shall contain the extent of the spatial domain of the coverage. It uses the value class *CV_GridEnvelope* which provides the grid coordinate values for the diametrically opposed corners of the grid. The [underline]#default is that this value is derived# from the bounding box for the data set or tile in a multi tile data set. In the encoding the property BoundingBox is used to hold the extent.
+The attribute _extent_ has the value class *CV_GridEnvelope* that shall contain the extent of the spatial domain of the coverage. It uses the value class *CV_GridEnvelope* which provides the grid coordinate values for the diametrically opposed corners of the grid. The [underline]#default is that this value is derived# from the bounding box for the data set or tile in a multi tile data set. The extent of the dataset limits is defined as a rectangle coincident with the outermost cell boundaries of the dataset. In the encoding the property BoundingBox is used to hold the extent.
```
[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/04-data_content_struct.adoc:104]: I'm not a native english speaker either, but I see many: northern limit, southern limit, western limit and eastern limit. And these combined create the extent/bounding box.

[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/04-data_content_struct.adoc:104]: I agree.

File: sources/3.0.0/sections/04-data_content_struct.adoc [hunk 7]
```diff
@@ -157,12 +157,12 @@ The class *CV_GridEnvelope* provides the grid coordinate values for the diametri
 [level=7]
 ===== low
 
-The attribute _low_ shall be the minimal coordinate values for all grid points within the envelope. For this specification this represents the Southwestern coordinate.
+The attribute _low_ shall be the minimal coordinate values for all grid points within the envelope. For this specification this represents the Southwestern corner coordinate of the southwesternmost grid cell.
```
[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/04-data_content_struct.adoc:160]: A clarification of some sort might be useful. As grid points are in the center of the cell this would be grid point coordinate minus half grid spacing in both X- and Y-directions.

File: sources/3.0.0/sections/04-data_content_struct.adoc [hunk 8]
```diff
@@ -157,12 +157,12 @@ The class *CV_GridEnvelope* provides the grid coordinate values for the diametri
...
 
 [level=7]
 ===== high
 
-The attribute _high_ shall be the maximal coordinate values for all grid points within the envelope. For this specification this represents the Northeastern coordinate.
+The attribute _high_ shall be the maximal coordinate values for all grid points within the envelope. For this specification this represents the Northeastern corner coordinate of the northeasternmost grid cell.
```
[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/04-data_content_struct.adoc:165]: A clarification of some sort might be useful. As grid points are in the center of the cell this would be grid point coordinate plus half grid spacing in both X- and Y-directions.

File: sources/3.0.0/sections/04-data_content_struct.adoc [hunk 9]
```diff
@@ -229,17 +229,23 @@ In S-102 there are currently no complex attributes defined.
...
+The grid properties of origin and spacing are defined by attributes in the *BathymetryCoverage.01* Feature Container Group. The grid is a two-dimensional matrix organized in row major order and starting from the southwestern-most grid cell. Thus, the first sample of the grid is the grid point at the southwest corner of the grid with location specified by the georeferencing parameters, the second is one grid resolution unit to the east of that position and at the same northing or latitude, and the third is two grid resolution units to the east and at the same northing or latitude. For stem:[C] columns in the grid, the stem:[(C+1)]^th^ sample in the grid is located one grid resolution unit to the north but on the same easting or longitude as the first sample in the grid.
 
 [[fig-s102-grid-node-location]]
-.S-102 Grid Node location
+.S-102 Grid point location
```
[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/04-data_content_struct.adoc:248]: In the final parsed document the picture filename is not present. These changes were also not made to annex F as it's not going to be included in the document.
Having said that, maybe this is a little confusing. I'll rename the picture accordingly.

File: sources/3.0.0/sections/07-data_cap_class.adoc [hunk 10]
```diff
@@ -9,7 +9,7 @@ A number of sounding techniques are used to capture bathymetric data. It is perm
 
 Quality and source metadata in S-102 are intended to enable and support future navigation software to appropriately auto-generate and attribute cartographic features such as custom depth contours and soundings from S-102 products, all while minimally impacting the overall file size of the product.
 
-Quality and source metadata are encoded in a raster attribute table that is compliant with HDF-5 and S-100 and will provide valuable information about the bathymetry on a node-by-node basis compared to traditional vector-based metadata files, simplifying the interpretation and implementation by navigation software systems.
+Quality and source metadata are encoded in a raster attribute table that is compliant with HDF-5 and S-100 and will provide valuable information about the bathymetry on a cell-by-cell basis compared to traditional vector-based metadata files, simplifying the interpretation and implementation by navigation software systems.
```
[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/07-data_cap_class.adoc]: Done. I replaced cell-by-cell with more precise "grid cell".

File: sources/3.0.0/sections/10-data_product_format.adoc [hunk 11]
```diff
@@ -639,10 +639,17 @@ For example, "latitude,longitude". Reverse scan direction along an axis is indic
 
 | 8
 | Interpolation type
-| interpolationType
+| nearestNeighbor
```
[the remark this answers is not in the document]
Comment (AnnaWall01) [reply sources/3.0.0/sections/10-data_product_format.adoc:642]: SMA agree on your change @RohdeBSH. Can I just click "Commit suggestion" or will this automatically be changed when merged?

File: sources/3.0.0/sections/10-data_product_format.adoc (hunk 4 above)
Comment (AnnaWall01) [reply sources/3.0.0/sections/10-data_product_format.adoc:652]: We agree that this might be a bit superfluous and with Topi's picture that shows where the grid point is placed, in the center of the cell, this becomes much clearer. We initially added this after dialogue with Raphael in trying to solve the cell area vs node issue. We do not see the need to add this attribute, as long as the cell area definition has been handled in the way we intend.

File: sources/3.0.0/sections/04-data_content_struct.adoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (AnnaWall01) [reply sources/3.0.0/sections/04-data_content_struct.adoc:104]: We agree with Topi, but perhaps an English speaking PT member can double check this for us (@hasel001). Also connected to section 12, below.

File: sources/3.0.0/sections/12-metadata.adoc (hunk 5 above)
Comment (AnnaWall01) [reply sources/3.0.0/sections/12-metadata.adoc:404]: Can an English speaking PT member double check this for us (@hasel001). This one is connected to section 4, above.

File: sources/3.0.0/sections/04-data_content_struct.adoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (AnnaWall01) [reply sources/3.0.0/sections/04-data_content_struct.adoc:104]: Agree. I can try and make the change right away.

File: sources/3.0.0/sections/04-data_content_struct.adoc [hunk 12]
```diff
@@ -229,17 +229,23 @@ In S-102 there are currently no complex attributes defined.
 === Dataset types
 
 ==== Introduction
-Bathymetric Surface datasets are represented as a discrete array of points contained in a regular grid. The general structure for a regular grid is defined in <<iho-s100,part=8>>.
+Bathymetric Surface datasets are represented as regular grids. The general structure for a regular grid is defined in <<iho-s100,part=8>>.
+
+The geo-referencing for an S-102 Bathymetric Surface product shall be grid point based, referenced from the southwestern-most grid point. Each sample in a grid represents the value in the grid cell. The reference position included in the metadata shall be given in the coordinates used for the grid and shall contain sufficient digits of precision to locate the grid with accuracy no worse than a decimetre on the surface of the ellipsoid of rotation of the chosen horizontal datum.
```
[the remark this answers is not in the document]
Comment (AnnaWall01) [reply sources/3.0.0/sections/04-data_content_struct.adoc:234]: This part comes from section 5.1 and is an old text. Is quite confusing and we tried to only make changes for this specific issue regarding cell area. If it makes it somewhat clearer with adding the attributes you mention, we are all for it.

File: sources/3.0.0/sections/04-data_content_struct.adoc (hunk 7 above)
[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/04-data_content_struct.adoc:160]: That sounds like a double minus/plus to me. To me personally the formatting is clear as it is, but if we'd like to have this clarified further we should probably write something like:

[Low] ...for this specification the value is southwesternmost _grid point_ coordinate _minus half grid spacing_ for each coordinate axis

[High] ...for this specification the value is northeasternmost _grid point_ coordinate _plus half grid spacing_ for each coordinate axis

What do you think?

File: sources/3.0.0/sections/04-data_content_struct.adoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (hasel001) [reply sources/3.0.0/sections/04-data_content_struct.adoc]: I agree with "limits". 

I can certainly see it from both perspectives. Although I don't think we are limited to 1 reasonable answer, we must certainly be internally consistent. Looking back as far as Ed. 2.1.0, it appears we have used "limit" to refer to scalar values (e.g., file size) and "limits" to refer to vector quantities (e.g., the spatial boundary enveloping a dataset). Let's go with limits.

File: sources/3.0.0/sections/12-metadata.adoc (hunk 5 above)
Comment (hasel001) [reply sources/3.0.0/sections/12-metadata.adoc:404]: I agree with "limits" (as above).

I can certainly see it from both perspectives. Although I don't think we are limited to 1 reasonable answer, we must certainly be internally consistent. Looking back as far as Ed. 2.1.0, it appears we have used "limit" to refer to scalar values (e.g., file size) and "limits" to refer to vector quantities (e.g., the spatial boundary enveloping a dataset). Let's go with limits.

File: sources/3.0.0/sections/03-data_product_id.adoc [hunk 13]
```diff
@@ -4,7 +4,7 @@
 
 *Title*:: Bathymetric Surface
 
-*Abstract*:: The Bathymetric Surface Product consists of a set of values organized to form a regular set of grid coverages, with associated metadata, for an area of the sea, river, lake, or other body of water. The final grid coverages include a depth value and optional associated uncertainty estimate for each location in the matrix. 
+*Abstract*:: The Bathymetric Surface Product consists of a set of values organized to form a regular grid coverage, with associated metadata, for an area of the sea, river, lake, or other body of water. The final grid coverages include a depth value and optional associated uncertainty estimate for each cell in the grid.
```
[the remark this answers is not in the document]
Comment (hasel001) [reply sources/3.0.0/sections/03-data_product_id.adoc:7]: I agree with Daniel that the number should agree between the first and second sentence (not necessarily from a grammar standpoint; instead, more from a semantic one). Furthermore, unless calling it one coverage would create some ambiguity that I am not foreseeing, I think that's the best option. The passage would then read:

*Abstract*:: The Bathymetric Surface Product consists of a set of values organized to form a regular grid coverage, with associated metadata, for an area of the sea, river, lake, or other body of water. The final grid coverage includes a depth value and optional associated uncertainty estimate for each cell in the grid.

File: sources/3.0.0/sections/04-data_content_struct.adoc (hunk 7 above)
[the remark this answers is not in the document]
Comment (hasel001) [reply sources/3.0.0/sections/04-data_content_struct.adoc:160]: I agree with @tfilppula in general but would like to propose a very slight change to improve readability. Including the context in @AnnaWall01 's suggestion above:

The attribute _low_ shall be the minimum coordinate values for each coordinate axis among all grid cells within the envelope. For this specification the value is the southwesternmost grid point coordinate minus half of the grid spacing for each coordinate axis.

The attribute _high_ shall be the maximum coordinate values for each coordinate axis among all grid cells within the envelope. For this specification the value is the northeasternmost grid point coordinate plus half of the grid spacing for each coordinate axis.

**Review by hasel001 — APPROVED** [review]
I agree with the changes in general. Obviously if you implement my suggestions in the comments (coverage, limit, limit, and low/high) I am in favor of those. But even if you resolve those issues in some other sensible way, I still approve.

File: sources/3.0.0/sections/10-data_product_format.adoc
Comment (avautour) [inline sources/3.0.0/sections/10-data_product_format.adoc:1]: So, coverages usually have two conventions, pixel-is-point and pixel-is-area. S-102 was based on BAG, which is pixel-is-point, and if I understand this pull request correctly, this is to change S-102 to pixel-is-area.

For pixel-is-point, the origin is usually a corner node (centre of the cell), while for pixel-is-area, the origin is usually the corner of a cell. As far as I can tell, line 771 of this file still states the the origin is at the centre of the cell, is that intentional? In other words, is the plan to use a pixel-is-point convention for the origin, and a pixel-is-area convention for the bounding box, because I would find that very weird.

For instance, see GeoTIFF, that has both conventions: https://docs.ogc.org/is/19-008r4/19-008r4.html#RasterSpace

File: sources/3.0.0/sections/03-data_product_id.adoc (hunk 13 above)
[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/03-data_product_id.adoc]: Changed according to @hasel001 comment.
@RohdeBSH : Can this be marked as resolved?

File: sources/3.0.0/sections/04-data_content_struct.adoc (hunk 7 above)
[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/04-data_content_struct.adoc]: Changed according to @hasel001 proposal.

File: sources/3.0.0/sections/04-data_content_struct.adoc (hunk 8 above)
[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/04-data_content_struct.adoc]: Marking this as resolved since this is discussed above.

File: sources/3.0.0/sections/10-data_product_format.adoc (hunk 11 above)
[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/10-data_product_format.adoc]: @RohdeBSH : I changed this to

`
| 8
| Interpolation type
| interpolationType
| 1
| Enumeration
| Value: 1 (nearestneighbor). See <<iho-s100,part=10c,table=22>>
`

[the remark this answers is not in the document]
Comment (RohdeBSH) [reply sources/3.0.0/sections/10-data_product_format.adoc]: > SMA agree on your change @RohdeBSH. Can I just click "Commit suggestion" or will this automatically be changed when merged?

I don't know. It's done anyway.

File: sources/3.0.0/sections/03-data_product_id.adoc (hunk 13 above)
[the remark this answers is not in the document]
Comment (RohdeBSH) [reply sources/3.0.0/sections/03-data_product_id.adoc]: @tfilppula 
Sorry, I thought Lawrence's version was already incorporated. Unfortunately, that was not the case.
So I tried to incorporate it myself, but unfortunately I got an error message in GitHub.
Can you please edit this? 
After that, it is complete.

[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/03-data_product_id.adoc]: @RohdeBSH 
This is done per Chair's comment. What we see in this discussion is outdated. Currently this part in Annas fork and pull request is:
"*Abstract*:: The Bathymetric Surface Product consists of a set of values organized to form a regular grid coverage, with associated metadata, for an area of the sea, river, lake, or other body of water. The final grid coverage includes a depth value and optional associated uncertainty estimate for each cell in the grid. "

I'll mark this conversation as resolved.

File: sources/3.0.0/sections/10-data_product_format.adoc (hunk 2 above)
Comment (tfilppula) [reply sources/3.0.0/sections/10-data_product_format.adoc]: S-102PT16 decided to limit the allowed value to "2: low". This change has been made and is included in the pull request.
I'll mark this conversation as resolved.

File: sources/3.0.0/sections/04-data_content_struct.adoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/04-data_content_struct.adoc]: As there is nothing to change, I'll mark this conversation as resolved.

[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/04-data_content_struct.adoc]: This has been done. Marking this conversation as resolved.

File: sources/3.0.0/sections/04-data_content_struct.adoc (hunk 12 above)
[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/04-data_content_struct.adoc:234]: @RohdeBSH @hasel001 @AnnaWall01 
Could this be revised as follows:
"... The reference position included in the metadata, precisely the attributes _gridOriginLatitude_ and _gridOriginLongitude_, shall be given in the coordinates used for the grid ..."

I you agree, I can make the change right away.

File: sources/3.0.0/sections/10-data_product_format.adoc (hunk 4 above)
Comment (tfilppula) [reply sources/3.0.0/sections/10-data_product_format.adoc:653]: @AnnaWall01 @RohdeBSH 
Does this require a new issue? If so, will @AnnaWall01 create it?

File: sources/3.0.0/sections/03-data_product_id.adoc
[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/03-data_product_id.adoc:1]: @RohdeBSH 
Could you formulate this to a new issue, please?

[the remark this answers is not in the document]
Comment (RohdeBSH) [reply sources/3.0.0/sections/03-data_product_id.adoc:1]: done

File: sources/3.0.0/sections/04-data_content_struct.adoc (hunk 12 above)
[the remark this answers is not in the document]
Comment (RohdeBSH) [reply sources/3.0.0/sections/04-data_content_struct.adoc:234]: If gridOriginLatitude  & gridOriginLongitude are correct, you are welcome to change it.

File: sources/3.0.0/sections/10-data_product_format.adoc (hunk 4 above)
Comment (RohdeBSH) [reply sources/3.0.0/sections/10-data_product_format.adoc:653]: I don't need a new issue for it, as long as the text always refers to the same point.

File: sources/3.0.0/sections/04-data_content_struct.adoc (hunk 12 above)
[the remark this answers is not in the document]
Comment (tfilppula) [reply sources/3.0.0/sections/04-data_content_struct.adoc]: Made the change, marking conversation as resolved.

File: sources/3.0.0/sections/10-data_product_format.adoc (hunk 4 above)
Comment (tfilppula) [reply sources/3.0.0/sections/10-data_product_format.adoc]: I removed the attribute dataoffsetCode as it is optional and not necessary.
Marking this conversation as resolved.

File: sources/3.0.0/sections/10-data_product_format.adoc
Comment (tfilppula) [reply sources/3.0.0/sections/10-data_product_format.adoc:1]: Created a new issue #83 for any discussion on grid origin definition,
Marking this conversation as resolved.

**Comment by tfilppula** [comment]
@AnnaWall01 
This is redy for your approval (pending your review). After your approval this can be merged,

**Comment by mikan66** [comment]
Reference email from Anna pm 4 APR 2024, Notice period for Pull Request #73.  Some thoughts from our NIWC team.

We have some concerns that what is presented here deviates from the ISO 19123 /  OGC 07-011. 
I don’t have the ISO, but from OGC 07-011:
image

The grid origin is clearly defined as the 0,0 position for the entire grid, by fundamental definition.  There are other pieces to this puzzle that ripple out from this, but this is a foundational definition.  S-100 Edition 5.1 section 8-5.2.8 Grid cell structure, introduces “offsets” that further complicate and deviate from the source ISO specs. 

In S-100 this is captured in Table 10c-10 – Attributes of feature container groups,  "dataOffsetCode". But this is only in the ENCODING and we can’t find it in the S-100 general feature model.  The “dataOffsetCode” appears to only apply to the data values (grid points), not to the origin, but it is not entirely clear.  Anyway, in total, this makes the encoding specific to HDF5 products, we think.

As such, “dataOffsetCode” is a bad idea because it deviates from the ISO spec, but nonetheless it is in S-100 for HDF5 encoding only and you can technically use it.  I personally find it to further confuse the definitions of everything grid related, whereas the ISO spec clearly establishes the relationships.

In summary, from your email diagram below, the “X”, (grid origin), should be at X0, Y0 if you want to be maximally conformant to the ISO specs.
image

**Comment by tfilppula** [comment]
The comment above by @mikan66 is related to issue #83.
This issue was created based on earlier comment on this PR.

**Review by AnnaWall01 — APPROVED** [review]
